aws/resource_aws_launch_configuration_test.go

```diff
@@ -281,15 +281,37 @@ func TestAccAWSLaunchConfiguration_updateEbsBlockDevices(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckAWSLaunchConfigurationExists("aws_launch_configuration.baz", &conf),
 					resource.TestCheckResourceAttr(
-						"aws_launch_configuration.baz", "ebs_block_device.2764618555.volume_size", "9"),
+						"aws_launch_configuration.baz", "ebs_block_device.1393547169.volume_size", "9"),
```

Comment:
Is this okay that it changed?

Reply:
It worried me too when I saw that, but I did verify that state migration is not necessary:

```
$ terraform init && terraform apply && terraform plan

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + aws_launch_configuration.baz
      id:                                                <computed>
      associate_public_ip_address:                       "false"
      ebs_block_device.#:                                "1"
      ebs_block_device.2764618555.delete_on_termination: "true"
      ebs_block_device.2764618555.device_name:           "/dev/sdb"
      ebs_block_device.2764618555.encrypted:             "true"
      ebs_block_device.2764618555.iops:                  <computed>
      ebs_block_device.2764618555.snapshot_id:           <computed>
      ebs_block_device.2764618555.volume_size:           "9"
      ebs_block_device.2764618555.volume_type:           <computed>
      ebs_optimized:                                     <computed>
      enable_monitoring:                                 "true"
      image_id:                                          "ami-5189a661"
      instance_type:                                     "t2.micro"
      key_name:                                          <computed>
      name:                                              <computed>
      root_block_device.#:                               "1"
      root_block_device.0.delete_on_termination:         "true"
      root_block_device.0.iops:                          <computed>
      root_block_device.0.volume_size:                   "11"
      root_block_device.0.volume_type:                   "gp2"


Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_launch_configuration.baz: Creating...
  associate_public_ip_address:                       "" => "false"
  ebs_block_device.#:                                "" => "1"
  ebs_block_device.2764618555.delete_on_termination: "" => "true"
  ebs_block_device.2764618555.device_name:           "" => "/dev/sdb"
  ebs_block_device.2764618555.encrypted:             "" => "true"
  ebs_block_device.2764618555.iops:                  "" => "<computed>"
  ebs_block_device.2764618555.snapshot_id:           "" => "<computed>"
  ebs_block_device.2764618555.volume_size:           "" => "9"
  ebs_block_device.2764618555.volume_type:           "" => "<computed>"
  ebs_optimized:                                     "" => "<computed>"
  enable_monitoring:                                 "" => "true"
  image_id:                                          "" => "ami-5189a661"
  instance_type:                                     "" => "t2.micro"
  key_name:                                          "" => "<computed>"
  name:                                              "" => "<computed>"
  root_block_device.#:                               "" => "1"
  root_block_device.0.delete_on_termination:         "" => "true"
  root_block_device.0.iops:                          "" => "<computed>"
  root_block_device.0.volume_size:                   "" => "11"
  root_block_device.0.volume_type:                   "" => "gp2"
aws_launch_configuration.baz: Creation complete after 5s (ID: terraform-20180405195506238300000001)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

aws_launch_configuration.baz: Refreshing state... (ID: terraform-20180405195506238300000001)

------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.

$ terraform init && terraform plan

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

aws_launch_configuration.baz: Refreshing state... (ID: terraform-20180405195506238300000001)

------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.

$ terraform show
aws_launch_configuration.baz:
  id = terraform-20180405195506238300000001
  associate_public_ip_address = false
  ebs_block_device.# = 1
  ebs_block_device.2764618555.delete_on_termination = true
  ebs_block_device.2764618555.device_name = /dev/sdb
  ebs_block_device.2764618555.encrypted = true
  ebs_block_device.2764618555.iops = 0
  ebs_block_device.2764618555.snapshot_id =
  ebs_block_device.2764618555.volume_size = 9
  ebs_block_device.2764618555.volume_type =
  ebs_optimized = false
  enable_monitoring = true
  ephemeral_block_device.# = 0
  iam_instance_profile =
  image_id = ami-5189a661
  instance_type = t2.micro
  key_name =
  name = terraform-20180405195506238300000001
  root_block_device.# = 1
  root_block_device.0.delete_on_termination = true
  root_block_device.0.iops = 0
  root_block_device.0.volume_size = 11
  root_block_device.0.volume_type = gp2
  security_groups.# = 0
  spot_price =
  vpc_classic_link_id =
  vpc_classic_link_security_groups.# = 0


$ terraform refresh
aws_launch_configuration.baz: Refreshing state... (ID: terraform-20180405195506238300000001)

$ terraform show
aws_launch_configuration.baz:
  id = terraform-20180405195506238300000001
  associate_public_ip_address = false
  ebs_block_device.# = 1
  ebs_block_device.1393547169.delete_on_termination = true
  ebs_block_device.1393547169.device_name = /dev/sdb
  ebs_block_device.1393547169.encrypted = true
  ebs_block_device.1393547169.iops = 0
  ebs_block_device.1393547169.no_device = false
  ebs_block_device.1393547169.snapshot_id =
  ebs_block_device.1393547169.volume_size = 9
  ebs_block_device.1393547169.volume_type =
  ebs_optimized = false
  enable_monitoring = true
  ephemeral_block_device.# = 0
  iam_instance_profile =
  image_id = ami-5189a661
  instance_type = t2.micro
  key_name =
  name = terraform-20180405195506238300000001
  root_block_device.# = 1
  root_block_device.0.delete_on_termination = true
  root_block_device.0.iops = 0
  root_block_device.0.volume_size = 11
  root_block_device.0.volume_type = gp2
  security_groups.# = 0
  spot_price =
  vpc_classic_link_id =
  vpc_classic_link_security_groups.# = 0
```

You can see in the last step during `refresh` that index is automatically re-computed and plan does not show any diff in the previous step. I don't know the exact reason behind that, but I'm confident that we don't need state migration.